tests/unit/integrations/protocols/test_http_client.py

```diff
@@ -0,0 +1,369 @@
+"""Unit tests for HTTPClient protocol."""
+
+from unittest.mock import AsyncMock, Mock
+
+import httpx
+import pytest
+from pydantic import SecretStr
+
+from openhands.integrations.protocols.http_client import HTTPClient
+from openhands.integrations.service_types import (
+    AuthenticationError,
+    RateLimitError,
+    RequestMethod,
+    ResourceNotFoundError,
+    UnknownException,
+    User,
+)
+
+
+class TestableHTTPClient:
```

Comment:
@openhands please fix the mocking for this; make sure `TestableHTTPClient` is implementing the `HTTPClient` protocol so that we aren't duplicating so much code

also remove the abstract method require `get_user` from the `HTTPClient` protocol; the base mixins should remain unchanged though they should still implement `get_user`